pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,259 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+Implementation of the BaseRepository abstract classes that defines an in-memory
+repository based on dictionaries.
+"""
+
+from __future__ import absolute_import, print_function
+
+import six
+
+from pywbem import CIMClass, CIMQualifierDeclaration, CIMInstance
+
+from pywbem._nocasedict import NocaseDict
+from pywbem._utils import _format
+
+from ._baserepository import BaseObjectStore, BaseRepository
+from ._utils import _uprint
+
+__all__ = ['InMemoryRepository', 'InMemoryObjectStore']
+
+
+class InMemoryObjectStore(BaseObjectStore):
+    """
+    A store for CIM objects of a single type (CIM classes, CIM instances,
+    or CIM qualifier declarations) that maintains its data in memory.
+    """
+    # Documentation for the methods and properties inherited from
+    # ~pywbem_mock:`BaseObjectStore` is also inherited in the pywbem
+    # documentation. Therefore the methods in this class have no documentation
+    # string.
+
+    # pylint: disable=line-too-long
+    def __init__(self, cim_object_type):
```

Comment:
For some reason, the docstring of the bae class `__init__()` is not shown in the generated docs.

The base class `__init__()` does not have an ´@abstractmethod` decorator but then I'm not sure whether they are a special case w.r.t. that. If the decorator cannot be used, an alternative would of course be to repeat the decription.

Reply:
This one gets weird since it does include an implementation statement (setting the type). However, apparently the concept of abstract does allow this but the subclasses must use the superclass... construct which we do. 

I added the @abstractmethod to force subclasses to implement the __init__ method.

COMMENT: The Python abstract concept does leave some ambiguities.

Marking this as DONE

Reply:
This one gets weird since it does include an implementation statement (setting the type). However, apparently the concept of abstract does allow this but the subclasses must use the superclass... construct which we do. 

I added the @abstractmethod to force subclasses to implement the __init__ method.

COMMENT: The Python abstract concept does leave some ambiguities.

Marking this as DONE

Reply:
This change did not make it into the last commit.

Reply:
Wierd as I show it in the code.

    @abstractmethod
    def __init__(self, cim_object_type):

Reply:
I now see it as well, and it has been there before. I was fooled by the fact that I made the comment on the derived class, bt the change is on the base class. DONE.